File: scheduler/LatihanAlarmManager/java/app/src/main/java/com/dicoding/picodiploma/myalarmmanager/utils/TimePickerFragment.java
package com.dicoding.picodiploma.myalarmmanager.utils;


import android.app.Dialog;
import android.app.TimePickerDialog;
import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import android.widget.TimePicker;

import java.util.Calendar;

public class TimePickerFragment extends DialogFragment implements TimePickerDialog.OnTimeSetListener {

    private DialogTimeListener mListener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        mListener = (DialogTimeListener) context;
    }

    @Override
    public void onDetach() {
        super.onDetach();
        if (mListener != null) {
            mListener = null;
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        final Calendar calendar = Calendar.getInstance();
        // Jam hari ini
        int hour = calendar.get(Calendar.HOUR_OF_DAY);

        // Menit dari jam hari ini
        int minute = calendar.get(Calendar.MINUTE);

        // True untuk format 24 jam, false untuk format 12 jam (am / pm)
        boolean formatHour24 = true;

        return new TimePickerDialog(getActivity(), this, hour, minute, formatHour24);
    }

    @Override
    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
        mListener.onDialogTimeSet(getTag(), hourOfDay, minute);
    }

    public interface DialogTimeListener {
        void onDialogTimeSet(String tag, int hourOfDay, int minute);
    }
}

File: scheduler/LatihanAlarmManager/kotlin/app/src/main/java/com/dicoding/picodiploma/myalarmmanager/utils/TimePickerFragment.kt
package com.dicoding.picodiploma.myalarmmanager.utils


import android.app.Dialog
import android.app.TimePickerDialog
import android.content.Context
import android.os.Bundle
import android.widget.TimePicker
import androidx.fragment.app.DialogFragment
import java.util.*

class TimePickerFragment : DialogFragment(), TimePickerDialog.OnTimeSetListener {

    private var mListener: DialogTimeListener? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mListener = context as DialogTimeListener?
    }

    override fun onDetach() {
        super.onDetach()
        if (mListener != null) {
            mListener = null
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val calendar = Calendar.getInstance()
        // Jam hari ini
        val hour = calendar.get(Calendar.HOUR_OF_DAY)

        // Menit dari jam hari ini
        val minute = calendar.get(Calendar.MINUTE)

        // True untuk format 24 jam, false untuk format 12 jam (am / pm)
        val formatHour24 = true

        return TimePickerDialog(activity, this, hour, minute, formatHour24)
    }

    override fun onTimeSet(view: TimePicker, hourOfDay: Int, minute: Int) {
        mListener?.onDialogTimeSet(tag, hourOfDay, minute)
    }

    interface DialogTimeListener {
        fun onDialogTimeSet(tag: String?, hourOfDay: Int, minute: Int)
    }
}
